src/reanimated2/commonTypes.ts

```diff
@@ -8,13 +9,22 @@ export interface StyleProps extends ViewStyle, TextStyle {
   [key: string]: any;
 }
 
+interface SharedValueInternals<T> {
+  _isReanimatedSharedValue: true;
+  _animation?: AnimationObject | null;
+  _value?: T | Descriptor | AnimatableValue;
+}
+
 export interface SharedValue<Value> {
   value: Value;
-  addListener: (listenerID: number, listener: (value: any) => void) => void;
+  addListener: (listenerID: number, listener: (value: Value) => void) => void;
   removeListener: (listenerID: number) => void;
-  modify: (modifier?: (value: any) => any) => void;
+  modify: (modifier?: (value: Value) => unknown) => void;
```

Comment:
```suggestion
  modify: (modifier?: (value: Value) => Value | AnimatableValue) => void;
```

Reply:
This does not work. It breaks `InlineStylesTest`. If I understand this correctly, we are making here a little bit of circular dependency, in case of `Animated.View` `style` which is of type `StyleProp<ViewStyle>` needs to extend `Value`(as it is passed to `modifier`) and `Value` needs to extend `StyleProp<ViewStyle>`(as it's returned from `modifier`).

The problem is that when we do it like this
```
  function InlineStylesTest13() {
    const sv = useSharedValue({ transform: [{ translateX: 0 }] });
    return <Animated.View style={sv} />;
  }
```
type of `sv` is 

```
SharedValue<{
    transform: {
        translateX: number;
    }[];
}>
```
and `ViewStyle` is not compatible wit as `transform` in it is not required.

If we do it like this, it will work 
```
  function InlineStylesTest13() {
    const sv = useSharedValue<ViewStyle>({ transform: [{ translateX: 0 }] });
    return <Animated.View style={sv} />;
  }
```

but this requires users to specify the `ViewStyle` in each `SharedValue` they use for that, which is terrible.

That being said, I've found another solution xD